packages/x-date-pickers/src/internals/hooks/useDesktopPicker/useDesktopPicker.tsx

```diff
@@ -13,6 +13,13 @@ import { WrapperVariantContext } from '../../components/wrappers/WrapperVariantC
 import { BaseFieldProps } from '../../models/fields';
 import { PickersLayout } from '../../../PickersLayout';
 import { InferError } from '../validation/useValidation';
+import { buildWarning } from '../../utils/warning';
+
+const ariaLabelledByCantBeResolvedWarning = buildWarning([
```

Comment:
That's not strictly speaking a breaking change.
But it's something I am pretty sure will cause warning in a lot of application.
I would be interested to know how the core handles similar situations where people have a non accessible app.
Do we force them to be accessible with a warning, or do we let them be ?

@mnajdova maybe

Reply:
It could be a breaking change for someone, who also has configured testing framework to fail when encountering `console.warn` or `console.error`. 🤔

Reply:
Their is also the scenario of a developer being accessible thanks to other methods, such as `aria-label`. They would get the warning even if they do not need it.

I do no remember warning from material about accessibility. But all their components have an explanation about what should be provided to be accessible.

Maybe the solution is to do not use warning but encourage its usage thanks to  section in the documentation

Reply:
Very good point.
I also noticed that even in our docs we sometimes do not have a `label` provided and warning is thrown in such cases.
I'll remove the warnings and look for the best place to provide the documentation in. 👌

Reply:
I've added a section about a11y [here](https://github.com/mui/mui-x/pull/7608/files#diff-fa38598fe4ffcdee6ccec38dd96a24cef9faf2bb723808ad0af55f4c75a51571R86). What do you think?

Reply:
> Do we force them to be accessible with a warning, or do we let them be ?

I don't think we even warn for issues regarding a11y.

Reply:
@mnajdova Thank you for the confirmation.
We've updated the solution to include documentation section and removed the warnings. 👌